### PR TITLE
[fix](match query) fix array column match query failed without inverted index

### PR DESCRIPTION
**Pull request by Tanya-W**
## Proposed changes

fix array column match query without inverted index return error: Not supported input arguments types

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Tanya-W** [comment]
run buildall

**Comment by Tanya-W** [comment]
run buildall

File: be/src/vec/functions/match.cpp [hunk 1]
```diff
@@ -119,13 +162,28 @@ Status FunctionMatchAll::execute_match(const std::string& column_name,
...
+                                column_name, str_ref.to_string(),
+                                doris::segment_v2::InvertedIndexQueryType::MATCH_ALL_QUERY,
+                                inverted_index_ctx);
+                data_tokens.insert(data_tokens.end(), element_tokens.begin(), element_tokens.end());
+            }
+        } else {
+            const auto& str_ref = string_col->get_data_at(i);
+            data_tokens = doris::segment_v2::InvertedIndexReader::get_analyse_result(
+                    column_name, str_ref.to_string(),
+                    doris::segment_v2::InvertedIndexQueryType::MATCH_ALL_QUERY, inverted_index_ctx);
```
Comment (airborne12) [inline be/src/vec/functions/match.cpp:184]: Repeated code with MATCH_ANY?

**Comment by Tanya-W** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**